dev/com.ibm.ws.jsonb.cdi/src/com/ibm/ws/jsonb/cdi/JsonbExtension.java

```diff
@@ -0,0 +1,56 @@
+/*******************************************************************************
+ * Copyright (c) 2017 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.jsonb.cdi;
+
+import javax.enterprise.context.spi.CreationalContext;
+import javax.enterprise.event.Observes;
+import javax.enterprise.inject.spi.AfterBeanDiscovery;
+import javax.enterprise.inject.spi.BeanManager;
+import javax.enterprise.inject.spi.Extension;
+import javax.json.bind.Jsonb;
+import javax.json.bind.spi.JsonbProvider;
+import javax.json.spi.JsonProvider;
+
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+import com.ibm.websphere.ras.Tr;
+import com.ibm.websphere.ras.TraceComponent;
+import com.ibm.ws.cdi.extension.WebSphereCDIExtension;
+
+@Component(property = { "api.classes=javax.json.bind.Jsonb;com.ibm.ws.jsonb.cdi.JsonbBean" }, immediate = true)
+public class JsonbExtension implements Extension, WebSphereCDIExtension {
+
+    private static final TraceComponent tc = Tr.register(JsonbExtension.class);
+
+    @Reference
+    protected JsonbProvider jsonbProvider;
```

Comment:
Appears to be unused.  I think the create method shouldn't be static and obtaining providers via FrameworkUtil.  Instead, it should be using this field.

Reply:
Unfortunately we need to get these services through the svc registry, because the object instances that CDI invokes `afterBeanDiscovery` with will be an instances different than the one in the svc registry (i.e. no refs will be injected).  We do need to keep the unused refs in place however, so that it gates activation of this component on the presence of JSON-B and JSON-P providers.